bandwidth/bandwidth-lib.pl

```diff
@@ -8,26 +8,30 @@
 BEGIN { push(@INC, ".."); };
 use WebminCore;
 &init_config();
-if (&foreign_installed("syslog-ng", 1) == 2) {
+if (&has_command('journalctl')) {
+	$syslog_module = undef;
+	$syslog_journald = "journald" ;
+	}
+elsif (&foreign_installed("syslog-ng", 1) == 2) {
 	&foreign_require("syslog-ng");
 	$syslog_module = "syslog-ng";
 	}
 elsif (&foreign_installed("syslog")) {
 	&foreign_require("syslog");
 	$syslog_module = "syslog";
 	}
-else {
-	$syslog_module = undef;
-	}
 &foreign_require("cron", "cron-lib.pl");
 &foreign_require("net", "net-lib.pl");
 
 %access = &get_module_acl();
 
 $bandwidth_log = $config{'bandwidth_log'} || "/var/log/bandwidth";
-$hours_dir = $config{'bandwidth_dir'} || "$module_config_directory/hours";
+$hours_dir = $config{'bandwidth_dir'} ||
+	     (-e "$module_config_directory/hours" ?
+	     		"$module_config_directory/hours" :
+			"$module_var_directory/hours");
 $cron_cmd = "$module_config_directory/rotate.pl";
-$pid_file = "$module_config_directory/rotate.pid";
+$pid_file = "$module_var_directory/rotate.pid";
```

Comment:
isnt this going to break existing systems where rotate job is already running?

Reply:
Alright, thanks! Good catch, fixed now!